### ISSUE TITLE
Refactor GIE API into new Infer API with Gst Infer Engine (GIE) and Triton Infer Server (TIS) services

### ISSUE THREAD

**Issue by rjhowell44** [prominenceai/deepstream-services-library#532]

**Comment by rjhowell44** [comment]
The following existing GIE services
* `dsl_gie_primary_new`
* `dsl_gie_secondary_new`
* `dsl_gie_infer_config_file_get`
* `dsl_gie_infer_config_file_set`
* `dsl_gie_model_engine_file_get`
* `dsl_gie_model_engine_file_set`
* `dsl_gie_interval_get`
* `dsl_gie_interval_set`
* `dsl_gie_primary_pph_add`
* `dsl_gie_primary_pph_remove`

Will be updated to
* `dsl_infer_gie_primary_new`
* `dsl_infer_gie_secondary_new`
* `dsl_infer_config_file_get`
* `dsl_infer_config_file_set`
* `dsl_infer_gie_model_engine_file_get`
* `dsl_infer_gie_model_engine_file_set`
* `dsl_infer_interval_get`
* `dsl_infer_interval_set`
* `dsl_infer_primary_pph_add`
* `dsl_infer_primary_pph_remove`

Two new constructors for the Primary and Secondary Triton Inference Server
```C
/**
 * @brief creates a new, uniquely named Primary Triton Inference Server (TIS) object
 * @param[in] name unique name for the new TIS object
 * @param[in] infer_config_file pathspec of the Infer Config file to use
 * @param[in] interval frame interval to infer on. 0 = every frame, 
 * @return DSL_RESULT_SUCCESS on success, DSL_RESULT_TIS_RESULT otherwise.
 */
DslReturnType dsl_infer_tis_primary_new(const wchar_t* name, const wchar_t* infer_config_file,
    uint interval);

/**
 * @brief creates a new, uniquely named Secondary TIS object
 * @param[in] name unique name for the new TIS object
 * @param[in] infer_config_file pathspec of the Infer Config file to use
 * @param[in] infer_on_tis name of the Primary or Secondary TIS to infer on
 * @param[in] interval frame interval to infer on. 0 = every frame, 
 * @return DSL_RESULT_SUCCESS on success, DSL_RESULT_INFER_RESULT otherwise.
 */
DslReturnType dsl_infer_tis_secondary_new(const wchar_t* name, const wchar_t* infer_config_file,
    const wchar_t* infer_on_tis, uint interval);
```

**Comment by rjhowell44** [comment]
Merged into v0.16.alpha
